### PR TITLE
refactor: use Tuple instead of Struct in saw-remote-api backend

### PR DESCRIPTION
**Pull request by pnwamk**
Use the more generic "tuple" instead of "struct" notion of value in the server backend (while keeping "struct" in the llvm-interface). See https://github.com/GaloisInc/saw-script/issues/1031

### CODE REVIEW

**Comment by RyanGlScott** [comment]
Does this imply that we should rename the data structures corresponding to the `struct` type (in addition to the `struct` setup value)?

**Comment by pnwamk** [comment]
> Does this imply that we should rename the data structures corresponding to the `struct` type (in addition to the `struct` setup value)?

I'm re-reading @atomb's original [comments](https://github.com/GaloisInc/saw-script/pull/1026#discussion_r562038590) on the matter. The impression I got was the things which are not inherently LLVM-related should prefer the more generic "tuple", while things that _are_ LLVM specific should be free to use the terminology/type names/etc from LLVM. To that end, I _think_ it's right to rename the `SetupVal` to "tuple" because it is a crucible/core concept, whereas the [types](https://github.com/GaloisInc/saw-script/blob/5e3bff782195c4438ea64f870d60657e7196955a/saw-remote-api/src/SAWServer/Data/LLVMType.hs#L43) are LLVM specific, so they should keep "struct" as the name.

File: saw-remote-api/src/SAWServer/Data/SetupValue.hs [hunk 1]
```diff
@@ -44,7 +44,7 @@ instance FromJSON cryptolExpr => FromJSON (CrucibleSetupVal cryptolExpr) where
           TagNullValue -> pure NullValue
           TagCryptol -> CryptolExpr <$> o .: "expression"
           TagArrayValue -> ArrayValue <$> o .: "elements"
-          TagStructValue -> StructValue <$> o .: "fields"
+          TagTupleValue -> TupleValue <$> o .: "elements"
```
Comment (RyanGlScott) [inline saw-remote-api/src/SAWServer/Data/SetupValue.hs:47]: I know that this isn't the officially sanctioned source of documentation anymore, but I'd find it helpful if [`old-SAW.rst`](https://github.com/GaloisInc/saw-script/blob/665a08ff23b2740320d7ceb04bb8e183f625a6be/saw-remote-api/docs/old-Saw.rst) were updated to reflect this change to the SAW remote API, especially in light of #1121.

**Review by atomb — APPROVED** [review]
This looks good to me. And, yes, the idea was that language-agnostic stuff would use tuple/record names whereas language-specific stuff would use language-specific names (like `struct`). At the moment, that distinction is somewhat academic, but once the Java bindings are in place I think it'll be more meaningful.

**Pull request merged**